packages/generate/test/select.test.ts

```diff
@@ -1010,433 +1015,358 @@ SELECT __scope_0_defaultPerson {
-  expect(result?.title).toEqual("The Avengers");
-});
+  test("portable shape", async () => {
+    const baseShape = e.shape(e.Movie, (movie) => ({
+      ...movie["*"],
+    }));
+    const query = e.select(e.Movie, (m) => {
+      return {
+        ...baseShape(m),
+        characters: { name: true },
+        filter_single: e.op(m.title, "=", "The Avengers"),
+      };
+    });
 
-test("filter_single exclusive prop", async () => {
-  const query = e.select(e.Movie, () => ({
-    title: true,
-    filter_single: { title: "The Avengers" },
-  }));
-  const result = await query.run(client);
-  expect(result?.title).toEqual("The Avengers");
-});
+    const result = await query.run(client);
+    assert.ok(result?.rating);
+    assert.ok(result?.characters);
+  });
 
-test("filter_single composite", async () => {
-  const query = e.select(e.Movie, () => ({
-    title: true,
-    filter_single: { title: "The Avengers", release_year: 2012 },
-  }));
-  const result = await query.run(client);
-  expect(result?.title).toEqual("The Avengers");
-});
+  test("filter_single id", async () => {
+    const query = e.select(e.Movie, () => ({
+      title: true,
+      filter_single: { title: "The Avengers" },
+    }));
+    const result = await query.run(client);
+    assert.equal(result?.title, "The Avengers");
+  });
 
-test("filter_single composite truple", async () => {
-  const query = e.select(e.Profile, () => ({
-    slug: true,
-    filter_single: {
-      a: "adsf",
-      b: "adsf",
-      c: "adsf",
-    },
-  }));
+  test("filter_single exclusive prop", async () => {
+    const query = e.select(e.Movie, () => ({
+      title: true,
+      filter_single: { title: "The Avengers" },
+    }));
+    const result = await query.run(client);
+    assert.equal(result?.title, "The Avengers");
+  });
 
-  await query.run(client);
-});
+  test("filter_single composite", async () => {
+    const query = e.select(e.Movie, () => ({
+      title: true,
+      filter_single: { title: "The Avengers", release_year: 2012 },
+    }));
+    const result = await query.run(client);
+    assert.equal(result?.title, "The Avengers");
+  });
 
-test("filter_single expect error", async () => {
-  // @ts-expect-error
-  e.select(e.Movie, () => ({
-    title: true,
-    filter_single: { genre: e.Genre.Horror },
-  }));
-});
+  test("filter_single composite truple", async () => {
+    const query = e.select(e.Profile, () => ({
+      slug: true,
+      filter_single: {
+        a: "adsf",
+        b: "adsf",
+        c: "adsf",
+      },
+    }));
 
-test("filter_single card mismatch", async () => {
-  const query = e.select(e.Movie, (movie) => ({
-    title: true,
-    filter_single: e.op(movie.genre, "=", e.Genre.Action),
-  }));
+    await query.run(client);
+  });
 
-  await expect(query.run(client)).rejects.toThrow(CardinalityViolationError);
-});
+  test("filter_single expect error", async () => {
+    // @ts-expect-error filter_single only types exclusive props
```

Comment:
eslint wants you to write a "reason" for the `ts-expect-error` clauses, which end up serving a similar function as the `assert` functions' message argument, although you don't see them. Still, kinda nice.